tests/e2e/restart_test.go

```diff
@@ -21,39 +21,59 @@ package e2e
 
 import (
 	. "gopkg.in/check.v1"
+	"strings"
+	"log"
 
 	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
 	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
 	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
 	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
+	adminconst "github.com/vmware/docker-volume-vsphere/tests/constants/admincli"
+	adminutil "github.com/vmware/docker-volume-vsphere/tests/utils/admincli"
 )
 
 type RestartTestData struct {
 	config        *inputparams.TestConfig
 	volumeName    string
-	containerName string
+	containerName []string
 }
 
 func (s *RestartTestData) SetUpSuite(c *C) {
 	s.config = inputparams.GetTestConfig()
 	if s.config == nil {
+		log.Printf("Setup skipped")
 		c.Skip("Unable to retrieve test config, skipping docker/plugin restart tests")
 	}
 
 	s.volumeName = inputparams.GetUniqueVolumeName("restart_test")
-	s.containerName = inputparams.GetUniqueContainerName("restart_test")
+	s.containerName = []string{inputparams.GetUniqueContainerName("restart_test"),
+				inputparams.GetUniqueContainerName("restart_test"),
+				inputparams.GetUniqueContainerName("restart_test")}
 
 	// ensure there are no remanants from an earlier run
-	out, err := dockercli.RemoveContainer(s.config.DockerHosts[1], s.containerName)
+	out, err := dockercli.RemoveContainer(s.config.DockerHosts[1], s.containerName[0])
 
 	// Create a volume
 	out, err = dockercli.CreateVolume(s.config.DockerHosts[1], s.volumeName)
 	c.Assert(err, IsNil, Commentf(out))
+
+	// Add access for the second datastore to the VM's vmgroup
+	out, err = adminutil.AddDatastoreToVmgroup(s.config.EsxHost, adminconst.DefaultVMgroup, s.config.Datastores[1])
```

Comment:
Here we don't know config has two datastores.

input param valids a test config if it has atleast one datastore, not two. Quoting the check below
```
	if config.DockerHostNames[0] == "" || config.DockerHostNames[1] == "" ||
		len(config.Datastores) <= 1 {
		return nil
	}
```